configs/rust/targets/x86_64-bottlerocket-linux-gnu.json

```diff
@@ -1,14 +1,18 @@
 {
   "arch": "x86_64",
   "cpu": "x86-64",
-  "crt-objects-fallback": "false",
   "crt-static-respected": true,
   "data-layout": "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128",
   "dynamic-linking": true,
   "env": "gnu",
   "has-rpath": true,
   "has-thread-local": true,
-  "linker-flavor": "gnu-cc",
+  "link-self-contained": {
+    "components": [
+      "linker"
+    ]
+  },
+  "linker-flavor": "gnu-lld-cc",
```

Comment:
I want to continue using our cross linker for now, so let's try updating the `jq` invocation in the `update-rust.sh` script to:
1. delete the `link-self-contained` dict
2. clobber `linker-flavor` to always be `gnu-cc`